src/Microsoft.Diagnostics.EventFlow.ServiceFabric/ServiceFabricDiagnosticPipelineFactory.cs

```diff
@@ -34,8 +40,53 @@ public static DiagnosticPipeline CreatePipeline(string healthEntityName, string
 
             ConfigurationBuilder configBuilder = new ConfigurationBuilder();
             configBuilder.AddJsonFile(configFilePath);
-            IConfigurationRoot configurationRoot = configBuilder.Build();
+            configBuilder.AddServiceFabric(ConfigurationPackageName);
+            IConfigurationRoot configurationRoot = configBuilder.Build().ApplyFabricConfigurationOverrides(healthReporter);
+
             return DiagnosticPipelineFactory.CreatePipeline(configurationRoot, new ServiceFabricHealthReporter(healthEntityName));
         }
+
+        internal static IConfigurationRoot ApplyFabricConfigurationOverrides(this IConfigurationRoot configurationRoot, IHealthReporter healthReporter)
+        {
+            Debug.Assert(configurationRoot != null);
+            Debug.Assert(healthReporter != null);
+
+            Regex fabricValueReferenceRegex = new Regex(FabricConfigurationValueReference, RegexOptions.CultureInvariant, TimeSpan.FromMilliseconds(100));
+
+            // Use ToList() to ensure that configuration is fully enumerated before starting to modify it.
+            foreach (var kvp in configurationRoot.AsEnumerable().ToList())
+            {
+                if (kvp.Value == null)
+                {
+                    continue;
+                }
+
+                try
+                {
+                    Match valueReferenceMatch = fabricValueReferenceRegex.Match(kvp.Value);
+                    if (valueReferenceMatch.Success)
+                    {
+                        string valueReferencePath = ConfigurationPath.Combine(valueReferenceMatch.Groups["section"].Value, valueReferenceMatch.Groups["name"].Value);
+                        string newValue = configurationRoot[valueReferencePath];
+                        if (string.IsNullOrEmpty(newValue))
+                        {
+                            healthReporter.ReportWarning(
+                                $"Configuration value reference '{kvp.Value}' was encountered but no corresponding configuration value was found using path '{valueReferencePath}'", 
+                                EventFlowContextIdentifiers.Configuration);
+                        }
+                        else
+                        {
+                            configurationRoot[kvp.Key] = newValue;
+                        }
+                    }
+                }
+                catch (RegexMatchTimeoutException)
+                {
+                    continue;
```

Comment:
If the reg-match timed out, the exception will be swallowed. I am a bit concerned of that. I think we should at least give end user a hint on the exception. How about adding the exception message into the health report?